CDLUC3/dmsp_frontend_prototype
# added fix for broken signup and login test

## Description

When we worked on adding the header from production into this app, I lost the logout link that was already in place. I added it back.

Since this header isn't a final design, I haven't added any unit test. I just thought it would be helpful to have the "Login/Logout" buttons working.

Also, I fixed some issues with /signup and /login unit tests where it sometimes wasn't finding "global.fetch".

Fixes # ([103](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/103))

## Type of change

## How Has This Been Tested?
1) Click the "Login" button in the header to log in
2) After logging in, check that you are logged in by looking for "dmsp" cookie.
3) Click the "Logout" button in the header to log out
4) You should be redirected to /login page.


## Checklist:
- [NA] I have commented my code, particularly in hard-to-understand areas
- [NA] I have added tests that prove my fix is effective or that my feature works
- [NA] I have completed manual or automated accessibility testing for my changes
- [NA] New and existing unit tests pass locally with my changes
- [NA] Any dependent changes have been merged and published in downstream modules
